Sources/MoSoCore/MoSoUser.swift

```diff
@@ -2,12 +2,18 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import Foundation
+
 // TODO: this very simple implementation will change
 public class MoSoUser {
     public let username: String
     public let identifier: String
     @KeychainStorage public var session: MoSoSession?
-    @KeychainStorage public var pocketSession: MoSoSession?
+    @KeychainStorage public var pocketSession: MoSoSession? {
+        didSet {
+            NotificationCenter.default.post(name: .userAuthDidChange, object: nil)
```

Comment:
It might be worth adding the session as the object that's being sent with the notification. This way, we can appropriately (not) perform an action based on if the auth state is logged in or logged out. Something like this is mentioned in another comment: we might not want to make an apollo request if we know the user is logged out, for example.

Reply:
I'd like to keep MoSoSession provider as the one and only place to look for Auth. Instead of passing around instances, where we can start to run into threading issues.

How do you feel about that?

Reply:
It doesn't necessarily have to be an instance of a provider, just something reflective of the session. Alternatively, a logged in vs logged out notification. But, one is fine.

Reply:
This is an interesting thought, I'll sync up with you tomorrow about it!